### PR TITLE
doc: some sqlite db info for HACKING

### PR DESCRIPTION
**Pull request by jb55**

### CODE REVIEW

**Comment by cdecker** [comment]
We should probably also mention not to do this with `lightningd` running since queries will lock the DB and cause it to crash.

**Comment by jb55** [comment]
good call, I did not know this...

**Comment by jb55** [comment]
@cdecker is that only an issue with the repl? It seems weird that sqlite wouldn't support concurrent access, at least for read-only queries?

**Comment by practicalswift** [comment]
Concept ACK!

File: doc/HACKING.md [hunk 1]
```diff
@@ -98,6 +98,29 @@ will print out (to stderr) a command such as:
...
 to exit the infinite while loop, otherwise you can type `continue` to begin.
 
+Database
+--------
+
+c-lightning state is persisted in `lightning-dir`. It is a sqlite database
+stored in the `lightningd.sqlite3` file, typically under `~/.lightning`. You can
+run queries against this file like so:
+
+    $ sqlite3 ~/.lightning/lightningd.sqlite3 "SELECT hex(prev_out_tx), prev_out_index, status FROM outputs"
```
Comment (practicalswift) [inline doc/HACKING.md]: Nit: Uppercase to indicate SQL function. Terminate with `;`.

```
$ sqlite3 ~/.lightning/lightningd.sqlite3 "SELECT HEX(prev_out_tx), prev_out_index, status FROM outputs;"
```

File: doc/HACKING.md [hunk 2]
```diff
@@ -98,6 +98,29 @@ will print out (to stderr) a command such as:
...
+
+or you can launch into the sqlite3 repl and check things out from there:
```
Comment (practicalswift) [inline doc/HACKING.md]: Nit: `Or you can launch into the sqlite3 REPL and check things out from there:`

File: doc/HACKING.md [hunk 3]
```diff
@@ -98,6 +98,29 @@ will print out (to stderr) a command such as:
...
+    SQLite version 3.21.0 2017-10-24 18:55:49
+    Enter ".help" for usage hints.
+    sqlite> .tables
+    channel_configs  invoices         peers            vars
+    channel_htlcs    outputs          shachain_known   version
+    channels         payments         shachains
+    sqlite> .schema outputs
+    ...
+
+Some data is stored as raw bytes, use `hex(column)` to pretty print these.
```
Comment (practicalswift) [inline doc/HACKING.md]: Nit: Uppercase `HEX(column)` to indicate SQL function.

**Comment by practicalswift** [comment]
Good stuff. Added some minor nits above :-)

File: doc/HACKING.md (hunk 3 above)
Comment (jb55) [reply doc/HACKING.md]: :+1:

File: doc/HACKING.md (hunk 2 above)
Comment (jb55) [reply doc/HACKING.md]: :+1:

File: doc/HACKING.md (hunk 1 above)
Comment (jb55) [reply doc/HACKING.md]: I'm one of those weirdos who prefers lowercase sql but I'll update acknowledging I'm in the minority here.

**Comment by jb55** [comment]
updated

**Comment by practicalswift** [comment]
ACK 5f57c703e5ad8d8417f00f69aa1bca038328f331

**Comment by rustyrussell** [comment]
It seems weird to me too, but it's caused intermittent test failures :(

I think sqlite3 supports it, but it does reserve the right to fail commit, and require you to retry.  We don't do that, because we normally don't need to...

**Pull request merged**